packages/ketcher-macromolecules/src/components/contextMenu/SequenceItemContextMenu/helpers.ts

```diff
@@ -53,48 +81,54 @@ export const generateSequenceContextMenuProps = (
   if (!selections?.length) return;
 
   const selectionsFlatten: NodeSelection[] = flatten(selections);
+  const countOfSelections = selectionsFlatten.length;
+  const countOfNucleoelements = getCountOfNucleoelements(selectionsFlatten);
   let title: string;
-  let isSelectedAtLeastOneNucleotide = false;
-  let isSelectedOnlyNucleotides = true;
-  const selectionsCount = selectionsFlatten.length;
-  let isSequenceFirstsOnlyNucleotidesSelected = true;
+  let isSelectedAtLeastOneNucleoelement = false;
+  let isSelectedOnlyNucleoelements = true;
+  let isSequenceFirstsOnlyNucleoelementsSelected = true;
 
-  for (let i = 0; i < selectionsCount; i++) {
-    const node = selectionsFlatten[i].node;
-    if (node instanceof Nucleotide) {
-      const nodeAttachmentR1 = node.sugar.attachmentPointsToBonds.R1;
-      if (nodeAttachmentR1 !== null) {
-        isSequenceFirstsOnlyNucleotidesSelected = false;
-      }
-      isSelectedAtLeastOneNucleotide = true;
+  // Generate labeled elements for RNA Builder
+  const selectedSequenceLabeledNodes = generateLabeledNodes(selectionsFlatten);
+
+  for (let i = 0; i < selectedSequenceLabeledNodes.length; i++) {
+    const node = selectedSequenceLabeledNodes[i];
+    const prevNode = selectedSequenceLabeledNodes[i - 1];
+    const isNodeNucleotideOrNucleoside =
+      node.type === 'Nucleotide' || node.type === 'Nucleoside';
```

Comment:
I would propose to use enum here to change values in one place in future if needed

Reply:
Fixed, thanks!